src/deploy/functions/services/index.ts

```diff
@@ -2,17 +2,25 @@ import * as backend from "../backend";
 import * as iam from "../../../gcp/iam";
 import * as v2events from "../../../functions/events/v2";
 import { obtainStorageBindings, ensureStorageTriggerRegion } from "./storage";
+import { obtainFirebaseAlertsBindings, ensureFirebaseAlertsTriggerRegion } from "./firebaseAlerts";
 
 const noop = (): Promise<void> => Promise.resolve();
 
+interface Project {
+  projectId: string;
+  projectNumber: string;
+}
+
 /** A service interface for the underlying GCP event services */
 export interface Service {
   readonly name: string;
   readonly api: string;
 
   // dispatch functions
-  requiredProjectBindings: ((pId: any, p: any) => Promise<Array<iam.Binding>>) | undefined;
-  ensureTriggerRegion: (ep: backend.Endpoint & backend.EventTriggered) => Promise<void>;
+  requiredProjectBindings:
+    | ((project: Project, policy: iam.Policy) => Promise<Array<iam.Binding>> | Array<iam.Binding>)
```

Comment:
Since this type got really complicated, what if we instead keep the original type signature and instead have `obtainFirebaseAlertsBindings` return a promise instead? e.g.

```ts
export function obtainFirebaseAlertsBindings(
  project: { projectId: string; projectNumber: string },
  existingPolicy: iam.Policy
): Promise<Array<iam.Binding>> {
  // ...implementation
  return Promise.resolve([pubsubBinding]);
}
```

Reply:
fixed in the latest commit